Sources/Fuzzilli/Corpus/MarkovCorpus.swift

```diff
@@ -0,0 +1,150 @@
+import Foundation
+
+/// Corpus & Scheduler based on 
+/// Coverage-based Greybox Fuzzing as Markov Chain paper
+/// https://mboehme.github.io/paper/TSE18.pdf
+/// Simply put, the corpus keeps track of which paths have been found, and prioritizes seeds
+/// whose path has been hit less than average. Ideally, this allows the fuzzer to prioritize
+/// less explored coverage
+
+public class MarkovCorpus: ComponentBase, Corpus {
+
+    private var allIncludedPrograms: [Program] // All programs currently in the corpus
+    private var programExecutionQueue: [Program]  // Queue of programs to be executed next, all of which hit a rare edge
+
+    private var edgeMap: [UInt64:Program]
+
+    private var totalExecs: UInt64
+
+    public init() {
+        // The corpus must never be empty. Other components, such as the ProgramBuilder, rely on this
+        self.allIncludedPrograms = []
+        self.programExecutionQueue = []
+        self.totalExecs = 0
+        self.edgeMap = [:]
+        super.init(name: "Corpus")
+
+    }
+    
+    override func initialize() {
+        // The corpus must never be empty
+        if self.isEmpty {
+            for _ in 1...5 {
+                let b = fuzzer.makeBuilder()
```

Comment:
Maybe this code could be moved into the `Corpus` protocol via an extension since it's used by both Corpus implementations? E.g. `Corpus.makeSeedProgram()` or so?